api/app/services/training/unified_pipeline_service.py

```
@@ -1052,14 +1131,53 @@ async def approve_candidate(
+            logger.warning(
+                "Candidate %s links to missing FAQ %s; creating a new FAQ",
+                candidate.id,
+                candidate.faq_id,
+            )
+
+        try:
+            matches = self.faq_service.get_filtered_faqs(search_text=question)
+            for faq in matches:
+                if faq.question == question and faq.id:
+                    return faq.id
+        except Exception:
```

Ruff (0.15.20)

[warning] Do not catch blind exception: `Exception`

(BLE001)